### DISCUSSION TITLE
Knowlg Friday's - Community Connect

### DISCUSSION THREAD

**Discussion by vinukumar-vs** [Sunbird-Knowlg/Community#9] [General Discussions]
Folks,

Welcome to the first-ever Knowlg Fridays - a community connect event where we learn and grow together as a community! 

[![Knowlg Fridays](https://user-images.githubusercontent.com/4496096/168289802-b0912ce6-0c65-4b37-a398-5fecbe1d4df4.png)](https://forms.gle/vCvL6gTAXViN3Lbj7)
 
### Topics:

1.  **Asset Architecture**
     - How to define an asset and manage it using APIs?
     -  How the metadata of the assets sync b/w heterogeneous databases (Cassandra, Neo4j & EelasticSeacrch)
     
**Speakers**: @vinukumar-vs , @AmiableAnil  & @maheshkumargangula

**Join webinar - [Zoom link](https://us06web.zoom.us/j/82109956783?pwd=bjltelN4MHk2eHdpV1lGT29hbFlKUT09)**
Meeting ID: 821 0995 6783
Passcode: 002581

**Reference:**
[Rich & deverse assets](https://knowlg.sunbird.org/learn/capabilities/rich-and-diverse-assets)
[Content service architecture](https://knowlg.sunbird.org/learn/product-and-developer-guide/content-service/architecture)
[Asset search service](https://knowlg.sunbird.org/learn/product-and-developer-guide/assets-search-service)

  
@pramodkvarma, @mathewjpallan, @pkranga, @Prerna789, @rahul-kankaria, @coolbung, @aks30, @vijiurs, @ChakshuGautam, @kiranharidas187, @pritamps, @chitranshu-keshav @rahul101001000 @coolbung
@SanthoshVasabhaktula, @vrayulu, @surendrasinghs, @pallakartheekreddy, @krgauraw, @luckynvdu @reshmi-nair, @sharathkashyap, @gandham-santhosh, @kameshbhr,  @AmiableAnil, @Jayaprakash8887, @anandp504, @maheshkumargangula, @rhwarrier,

**Comment by pramodkvarma** [comment]
This is wonderful! 👏👏👏

**Comment by aks30** [comment]
@vinukumar-vs the registration form seems to be closed and not opened for all the public, do check once!

**Reply by Krishnaj20** [reply]
@aks30 - sorry for the inconvenience - Please try now.

**Comment by madhucr** [comment]
Great Initiative @vinukumar-vs @Krishnaj20

**Comment by surendrasinghs** [comment]
Superb!  👏🏽 Excited and registered for the session this Friday.

**Comment by Krishnaj20** [comment]
> The secret of getting ahead is getting started!

We had around 40 participants in today's session. Thank you all for joining in and making it an interactive session. We request you to share your feedback [here](https://docs.google.com/forms/d/12Rc32wpe3hOaLaD7lzu0uhWXxPcT65YMSqv9cnDueJc/edit?usp=sharing
).

In today’s session we have covered:
1) High level architecture of Sunbird Knowlg
2) Asset workflow with an example

Here is the link to the session recording:

https://us06web.zoom.us/rec/share/3VFqKEfHo5Ds25JKqn2bVvNQ0Nr8XBmS9K9ZDcX3kzHkXR8PP-0DGvIOlmZFOU42.lfYPcPrvCAJetRmV
Passcode: d!!?xN8&


[Here](https://docs.google.com/presentation/d/1OMpJjS1JlyawDdkn8L3eDIcRltpF1aPoyKIwVg8oe2g/edit?usp=sharing) is the link to the deck used in today’s session.

**Note: If you have any further questions about the topics covered in today’s session please post them in this thread itself.**

We will shortly publish the agenda for next week’s session!

**Comment by aks30** [comment]
Thank you @Krishnaj20 for organizing this event and @AmiableAnil , @vinukumar-vs for sharing and explaining the data workflow in Knowlg service.

I went through the recording as well as the deck and had a query:- Where do we define the ES schema corresponding to Neo4j schema? Is the Neo4j custom extension we use to generate a transactional event and the  search indexer job used to put data in ES, schema-agnostic i.e. works for all Neo4j schema

Also @Krishnaj20 in case you are still allowing volunteers to try hands-on for the next session, I would be happy to participate.

**Reply by Jayaprakash8887** [reply]
https://github.com/project-sunbird/sunbird-devops/tree/release-4.9.0/ansible/roles/es-mapping/files/mappings

**Reply by vinukumar-vs** [reply]
@aks30 Thanks for your feedback. We will be having a workshop session on the previous session(defining Schema & basic API's on schema properties).

We will share the ES indexer details soon or will try to cover in the next session.

**Comment by YasharthSingh** [comment]
@vinukumar-vs @Krishnaj20 We are trying to implement the architecture discussed for Cassandra, Neo4j, and ES. Just wanted to confirm that do we keep all the data that we have in Neo4j in ES as well or just the Keys most queried on?

**Reply by vinukumar-vs** [reply]
@YasharthSingh By default, all the data of Neo4J will sync to ES. 

For nested properties, we have to configure the content-service to sync as an object to perform ES queries on nested data.
